File: queue.c
#include<stdio.h>

#define Qsize 20
void enqueue(int*,int*);
void dequeue(int*,int*,int*);
void display(int*,int*,int*);
void main 
{ intq[Qsize],r=-1,f=0;
int choice2;
char choice='y';
clrscr()
while(choice1='y'// choice1='Y')
{ printf("\n enter choice of operation on queue:
          1.enqueue,2.dequeue,3.display\n")
scanf("%d",& choice2);
if(choice2=i)
{enqueue(q,&r);
display(q,&f,&r);
}
else if(choice2==2)
{dequeue(q,&f,&r)'
display(q,&f,&r);
}
else if(choice2==3)
{display(q,&f,&r);
}
else
printf("\n wrong input");
printf("\n want to continue? yes/no:enter y for yes & n for no:\t");
fflush(stdin);
scanf("%s",&choice);
}


void enqueue(int q[],int*r)
{ int item;
printf("enter item\n");
scanf("%d",& item);
if((*r==qsize-1))
printf("\n q is full");
+r==*r+1;
q[*r]=item;
printf("value inserted=%d",item);
}
void dequeue(int q[],int*f,int*r)
{
if(*f>*r);
{printf("queue is empty\n");
}
printf("element deleted is %d",q[*f]);
(*f)++;
}
display(int a[],int*f,int*r)
{int i;
if(*f>*r)
printf("no display");
for(i=*f;i<*r;i+);
{printf("%d",q[i]);
}

File: queue.java
#include <stdio.h>
#define max 10
int queue[max];
int front=-1,rear=-1;
void insert(void);
int  dequeue(void);
void display(void);
int main()
{
	int option;
	int ele;
	do
	{
		printf("\n\n *****main menu *****");
		printf("\n 1.insert an element");
		printf("\n 2.delete an element");
		printf("\n 3.display the queue");
		printf("\n 4.exit");
		printf("\n enter the option");
		scanf("%d",&option);
		switch(option)
		{
			case 1:insert();
			break;
			case 2:
			      ele=dequeue();
				  if(ele!=-1)
					  printf("\n the number deleted is:%d",&ele);
					  break;
			case 3:
                   display();
			       break;
		}
	}while(option !=5);
	return 0;
}
   void insert()
   {
	   int num;
	   printf("\n enter the number to be inserted in the queue:");
	   scanf("%d",&num);
	   if(rear==max-1)
		   printf("\n overflow");
		   else if(front==-1&&rear== -1)
			   front=rear=0;
			   else
				   rear++;
				   queue[rear]=num;
   }
   int dequeue()
   {
	   int ele;
	   if(front==-1 || front>rear)
	   {
		   printf("\n underflow");
		   return -1;
	   }
	   else
	   {
		   ele=queue[front];
		   front++;
		   if(front>rear)
			   front=rear=-1;
			   return ele;
	   }
   }
	     void display()
		 {
			 int i;
			 printf("\n");
			 if(front==-1 || front>rear)
				 printf("\n QUEUE IS EMPTY");
				 else
				 {
					 for(i=front;i<=rear;i++)
						 printf("\t %d",queue[i]);
				 }
		 }
